### DISCUSSION TITLE
unable to build vtm-android-example

### DISCUSSION THREAD

**Discussion by Meibes** [mapsforge/vtm#910] [Q&A] [answered]
hello,

i'm not sure if this problem is on my end only, but i can't build "vtm-android-example". i tried it on two different pc's with latest Android Studio, but i'm always getting errors like `error: package android.content does not exist import android.content.Context;`. i'm not sure if this is gradle-related, but looking through some gradle-files i stumbled upon `compileOnly files("${System.env.ANDROID_HOME}/platforms/android-${androidCompileSdk()}/android.jar")` - i'm not sure if the ANDROID_HOME has to be set manually, but after some digging a found a hint, that [ANDROID_HOME is deprecated](https://developer.android.com/studio/command-line/variables#envar) and the "local.properties"-file contains the path to the sdk.dir, which should be equal to ANDROID_HOME as far as i understand it.

cheers

Android Studio Bumblebee | 2021.1.1 Patch 2
Build #AI-211.7628.21.2111.8193401, built on February 17, 2022
Runtime version: 11.0.11+9-b60-7590822 amd64
VM: OpenJDK 64-Bit Server VM by Oracle Corporation
Windows 10 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 1280M
Cores: 8
Registry: external.system.auto.import.disabled=true
Non-Bundled Plugins: idea.plugin.protoeditor (2.3.1), org.jetbrains.kotlin (211-1.6.10-release-923-AS7442.40)

**Comment by devemux86** [comment, marked as answer]
Set an environment variable `ANDROID_HOME` with the path to the Android SDK installation directory.
https://developer.android.com/studio/command-line/variables

---

[GitHub Actions](https://github.com/mapsforge/vtm/actions) and command line build the repository successfully.

**Reply by Meibes** [reply]
it would be good to add this information to the README, because the environment variables are not set by default when installing android studio and this involves an extra step for the user. something like a BUILD-paragraph, informing the user to set `ANDROID_SDK_ROOT` e.g. `$ set ANDROID_SDK_ROOT=C:\Android\sdk\` instead of the deprecated `ANDROID_HOME`, to be able to build the project

**Comment by devemux86** [comment]
`ANDROID_SDK_ROOT` is deprecated:
https://developer.android.com/studio/command-line/variables

See also this discussion: https://github.com/mapsforge/mapsforge/discussions/1291
